.github/workflows/constraints.yml

```diff
@@ -13,8 +13,6 @@ jobs:
     steps:
       - name: Checkout main branch
         uses: actions/checkout@v4
```

Comment:
Yes. Though since we have to do that, we might as well do it _before_ the checkout action, so that `git` will be available for that action as well and it would not have to resort to using the GitHub API.

I’d also add the `--no-install-recommends` flag to `apt-get install`, to be sure it only installs `git` itself and what is strictly required to make it work, nothing more -- the entire point of using a bare Ubuntu here is to minimise the risk that Python packages in the base image could interfere with our process of installing our own packages, so we don’t want `apt-get install` deciding to happily install a bunch of Python packages somewhat related to `git` just because they are marked as “recommended with `git`”.

Reply:
Thanks, I have implemented that now, hopefully it looks ok to you.